phoenix-core/src/main/java/org/apache/phoenix/coprocessor/IndexRebuildRegionScanner.java

```diff
@@ -1191,6 +1203,9 @@ public boolean next(List<Cell> results) throws IOException {
         }
         Cell lastCell = null;
         int rowCount = 0;
+        if(!shouldRebuildOrVerify()) {
```

Comment:
This was one way without taking any additional parameters. I think the read repair won't be impacted as it would have different ts than the key of PIT result and this `if` will not get executed. We can wrap this under a parameter if you think that's the right thing to do.